src/commands/record.rs

```diff
@@ -30,6 +30,8 @@ use settings::RecordSettings;
 enum RecordFailure {
     #[fail(display = "unable to write to file: {}: file exists", path)]
     FileExists { path: String },
+    #[fail(display = "filename required when recording in raw mode")]
+    NoFileName,
```

Comment:
Let's also use `requires_if` in the `structopt` derive so we make sure this is always true. There is a similar but not quite the same example at https://github.com/TeXitoi/structopt/blob/master/examples/raw_attributes.rs#L30. The `clap` docs are at https://docs.rs/clap/2.26.2/clap/struct.Arg.html#method.requires_if

Reply:
Maybe `requires` is better?

Reply:
Ah, yep.